apps/theming/css/default.css

```
@@ -55,6 +55,7 @@
   --primary-invert-if-bright: no;
   --background-invert-if-dark: no;
   --background-invert-if-bright: invert(100%);
+  --invert-if-background-bright: no;
```

The names thou :P so close to confusion

---

Yeah, https://github.com/nextcloud/server/pull/34576 simplify the code a bit.

We should improve those variables imho

---

We should deprecate those soon enough anyway. They are here for compatibility purpose.
People should use inline svg instead.

Also, disappointed https://github.com/nextcloud/server/pull/33480 went in.
Why do we need `background-invert-if-bright` ? We should just provide icons that works staright away on light theme, so only the `invert-if-dark` is necessary

---

```suggestion
  --background-image-invert-if-bright: no;
```

---

Your suggestion even more makes it clear that the names are not easy to get.

Both are inverting background images of the applied element with a filter, but the old variable based on the background color, the new one based on the "theming/a11y" background image :D


```suggestion
  --background-invert-if-background-image-bright: no;
```

---

> Also, disappointed https://github.com/nextcloud/server/pull/33480 went in.

Sorry I have disappointed you. I just tried my API result which worked since Nextcloud 18 to not require providing an inverted image all of a sudden.

https://github.com/nextcloud/server/blob/9ae505cb7cdeb0519bdf4e9da577785fc1a655ce/lib/public/WorkflowEngine/IOperation.php#L54-L64

> We should just provide icons that works staright away on light theme, so only the invert-if-dark is necessary

yeah was a mistake back then, I guess mostly because all apps are mandatory shipping a white app icon already so it was used back then?

> People should use inline svg instead.

Would have worked in this case, but for APIs where clients are involved, it feels weird (also I'm not sure how easy it is for clients to turn an inline SVG into something they can render)

---

> Sorry I have disappointed you. I just tried my API result which worked since Nextcloud 18 to not require providing an inverted image all of a sudden.

My apologies, your fix was necessary. I wish things would be simpler, hard to try to fit all the requirements around here, legacy, fallbacks, etc etc :confounded: 
Again, sorry for my rough words, this was not aimed towards you :hugs:

---

Maybe even `invert-if-content-background-dark` to not confuse it with the option to picker a color as background instead of an image?

---

Changing variables names now is breaking though. Poor devs :'(

---

Well the last one is a new one? having a different pattern is bad, but is it worse then having one that can be confused with another one?

---

Would `--background-image-invert-if-bright` work as the new name then @nickvergessen @skjnldsv?